src/module.mk

```diff
@@ -135,6 +137,59 @@ endif
 LIBFLEXI_SRC := $(sort $(LIBFLEXI_SRC))
 LIBFLEXI_HDR := $(sort $(LIBFLEXI_HDR))
 
+# loop library #########################################################
+LOOP_DIR := $(DIR)/loop_libraries
+
+LOOP_SRC := \
+		$(LOOP_DIR)/loop_library.cpp
+
+LOOP_HDR := \
+		$(LOOP_DIR)/loop_library.hpp \
+		$(LOOP_DIR)/loop_library_interface.hpp
+
+LIBFLEXI_SRC += $(LOOP_SRC)
+LIBFLEXI_HDR += $(LOOP_HDR)
+
+ifeq ($(ENABLE_LOOPTOOLS),yes)
+LIBFLEXI_SRC += \
+		$(LOOP_DIR)/library_looptools.cpp
+LIBFLEXI_HDR += \
+		$(LOOP_DIR)/library_looptools.hpp
+endif
+
+ifeq ($(ENABLE_FFLITE),yes)
+LIBFLEXI_SRC += \
+		$(LOOP_DIR)/library_fflite.cpp
+LIBFLEXI_HDR += \
+		$(LOOP_DIR)/library_fflite.hpp
+endif
+
+ifeq ($(ENABLE_COLLIER),yes)
+LIBFLEXI_SRC += \
+		$(LOOP_DIR)/library_collier.cpp
+LIBFLEXI_HDR += \
+		$(LOOP_DIR)/library_collier.hpp
+
+$(LOOP_HDR) $(LOOP_SRC) : $(LOOP_DIR)/libcollier_wrapper.a
+
+$(LOOP_DIR)/libcollier_wrapper.a : $(LOOP_DIR)/collier_wrapper.o $(LOOP_DIR)/collier_wrapper.mod
+	@echo Building collier wrapper library
+	@ar cr $(LOOP_DIR)/libcollier_wrapper.a $(LOOP_DIR)/collier_wrapper.o
+
+$(LOOP_DIR)/collier_wrapper.mod $(LOOP_DIR)/collier_wrapper.o : $(LOOP_DIR)/collier_wrapper.f90
+ifneq ($(findstring gfortran,$(FC)),)
```

Comment:
That's a bit better, but not perfect yet. :)

The problem I see is that we have now *two* places where we check for the compiler. We should only have *one* place, though! This place should be the `configure` script.

I propose to rewrite this `if-else-endif` statement to use a single compilation unified statement, maybe like this:
~~~
@$(FC) $(FFLAGS) -c $(LOOP_DIR)/collier_wrapper.f90 -o $(LOOP_DIR)/collier_wrapper.o $(FMOD) $(LOOP_DIR)
~~~
The variables `FC` and `FFLAGS` are set at the beginning of the `Makefile`, which are generated by the `configure` script. In the `configure` script we shall distinguish between the different Fortran compilers and set the variables accordingly.

Reply:
@Expander, thank you for an advice! I think that the standard for a file should be collier specific, so I will not add it into ``FFLAGS`` but instead into ``COLLIERSTD``.